pkg/flatten/flatten.go

```diff
@@ -34,9 +34,14 @@ func Flatten(value interface{}, prefix string) map[string]types.String {
 			}
 		default:
 			val := "null"
-			if value != nil {
+			switch v := value.(type) {
+			case nil:
+			case float64:
+				val = strconv.FormatFloat(v, 'f', -1, 64)
+			default:
 				val = fmt.Sprintf("%v", value)
 			}
```

Comment:
This is already inside `switch value := value.(type)`, so shouldn't adding a `case float64:` there suffice?

Reply:
That would duplicate `flattened[key] = ...`.

Reply:
I see. The annoying part is the right hand side though, I'd consider just adding a helper function instead so that one could just write something like `flattened[key] = types.MakeString("null")`. That might benefit other places in the long run too as those nested struct initialization is just annoying.